include/mbedtls/config.h

```diff
@@ -256,6 +256,49 @@
  */
 //#define MBEDTLS_DEPRECATED_REMOVED
 
+/**
+ * \def MBEDTLS_CHECK_PARAMS
+ *
+ * This configuration controls whether the library validates parameters passed
+ * to it.
+ *
+ * Application code that deals with 3rd party input may wish to enable such
+ * validation, whilst code on closed systems, such as embedded systems, where
+ * the input is controlled and predictable, may wish to disable it entirely to
+ * reduce the code size of the library.
+ *
+ * When the symbol is not defined, no parameter validation except that required
+ * to ensure the integrity or security of the library are performed.
+ *
+ * When the symbol is defined, all parameters will be validated, and an error
+ * code returned where appropriate.
+ */
+#define MBEDTLS_CHECK_PARAMS
+
+/**
+ * \def MBEDTLS_PARAM_FAILED_CALLBACK
+ *
+ * Enable or disable parameter validation callback upon validation failure.
+ *
+ * The parameter validation failure may be handled in different ways. In most
+ * functions, the preferred reaction is to immediately return with an error
+ * code. However, some functions do not return an error code, but they take
+ * parameters, that need to be validated. In such case, Mbed TLS, besides
+ * returning from the function, will call a special failure callback function.
+ * The library provides a default implementation of that and allows supplying
+ * a user defined alternative.
+ *
+ * When this symbol is defined, the parameter validation callback will be
+ * compiled in and called upon the parameter validation failure.
+ *
+ * When this symbol is not defined, parameter validation will be limited to
+ * immediate returns of error codes, or just returns in the functions of void
+ * return type.
```

Comment:
@k-stachowiak @mpg @sbutcher-arm I am not sure that's a good design choice in case of functions without return value: We're not performing the requested operation but don't signal this in any way to the user? Let's assume for example the following: The function is an initialization function which is supposed to zeroize the structure first and then perhaps set some default values depending on the parameters passed. If some of these parameters is invalid and the function hence returns without performing any action, the user continues as if things went fine, subsequently calling the library with a potentially entirely corrupted structure (because it has never been initialized and hence be filled with garbage).